crates/rustic_core/examples/restore.rs

```diff
@@ -19,15 +17,14 @@ fn main() -> Result<(), Box<dyn Error>> {
     let node = repo.node_from_snapshot_path("latest", |_| true)?;
 
     // use list of the snapshot contents using no additional filtering
-    let recursive = true;
-    let streamer_opts = TreeStreamerOptions::default();
-    let ls = repo.ls(&node, &streamer_opts, recursive)?;
+    let streamer_opts = LsOptions::default();
+    let ls = repo.ls(&node, &streamer_opts)?;
```

Comment:
There currently is also `Repository::list` which lists all files of a given `FileType` in the repository. This is usually not needed by standard clients, but the rustic CLI has the `rustic list` command which exactly uses this. Therefore I wanted to use `repo::ls` here.

About additional methods using the standard options: This can be done, but I'm a but unaware of the use cases. For examples, it may be nice to get shorter examples. 
However, I think that in most real-life applications those options would be either anyway configurable by users or use a fixed config which IMO often is not the standard. So I think in most cases, programmers using the API would anyway want to use the methods with options. In that case I would even find it counter-productive to have other methods in the examples...